internal/store/job.go

```diff
@@ -302,6 +302,35 @@ var (
 					}
 				}
 
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_job_status_failed_reason",
+			Type: metric.Gauge,
+			Help: "The reason why the job failed its execution.",
+			GenerateFunc: wrapJobFunc(func(j *v1batch.Job) *metric.Family {
+				var ms []*metric.Metric
+
+				for _, condition := range j.Status.Conditions {
+
+					if condition.Type != v1batch.JobFailed {
+						continue
+					}
+
+					conditionMetrics := addConditionMetrics(condition.Status)
+
+					for _, m := range conditionMetrics {
+						metric := m
+						metric.LabelKeys = []string{"condition", "reason", "status"}
+						metric.LabelValues = append([]string{string(condition.Type), condition.Reason}, metric.LabelValues...)
```

Comment:
I would prefer this to be modeled on the pod reason we already have, this means we only allow the reasons
https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/pod.go#L38
https://github.com/kubernetes/kube-state-metrics/blob/master/internal/store/pod.go#L381

Reply:
@lilic - Should I remove the `condition, status` label keys from the metric? So, I should be only sending `reason` in the label keys, right? 

One more thing, I could only find `["BackoffLimitExceeded", "DeadLineExceeded"]` job failure reasons. `Evicted` seems to be a reason for pod failure. 
Could you please verify these findings?

Reply:
Yes the ones that are for the job failure, the pod reasons were just an example.

Reply:
@lilic I got your point. The problem is I'm not able to find the `Evicted` failure reason for the job in the Kubernetes job/controller code-base but there is an `Evicted` failure reason for the pod. In the [Issue 947](https://github.com/kubernetes/kube-state-metrics/issues/947) the reporter has given an example with `Evicted` reason. So, I just want to confirm whether there is an `Evicted` failure reason for the job or not.

Reply:
Let me have a look at the controller.

Reply:
If user reported I would add all 3, worst case it will just never get that reason, but since we see it is possible it most likely reflects it from the Pod Reason. So lets add two you found plus Evicted. Makes sense?

Reply:
Sounds good to me. I'll update the PR.